kahuna/public/js/edits/service.js

```diff
@@ -149,6 +149,17 @@ service.factory('editsService',
         return existingRequestPool.promise;
     }
 
+    function perform(resource, action, originalImage) {
+        const newRequest = resource.perform(action).
+              then(action => resource.get().then(edit => getSynced(originalImage, newImage => matches(edit, newImage))));
+
+        const existingRequestPool = updateRequestPools.get(resource) ||
+            registerUpdateRequest(resource, originalImage);
```

Comment:
I think they're pretty much one and the same thing - would it make more sense to just have a `requestPool`? What's the value in being able to differentiate between `update`s and `perform`s?